tuf/client/updater.py

```diff
@@ -1146,12 +1148,26 @@ def _update_root_metadata(self, current_root_metadata):
     # current = version 1
     # latest = version 3
     # update from 1.root.json to 3.root.json.
+
+    # Temporarily set consistent snapshot. Will be updated to whatever is set
+    # in the latest root.json after running through the intermediates with
+    # _update_metadata().
+    self.consistent_snapshot = True
+
     for version in range(next_version, latest_version + 1):
-      # Temporarily set consistent snapshot. Will be updated to whatever is set
-      # in the latest root.json after running through the intermediates with
-      # _update_metadata().
-      self.consistent_snapshot = True
       self._update_metadata('root', DEFAULT_ROOT_UPPERLENGTH, version=version)
```

Comment:
I somehow missed this line in my first review. It downloads all the root files again using `_get_metadata_file`, with all its verification routines I pointed out in my first review.